lib/Sema/TypeChecker.h

```diff
@@ -158,8 +158,12 @@ enum class NameLookupFlags {
   /// Whether to include results from outside the innermost scope that has a
   /// result.
   IncludeOuterResults = 1 << 1,
-  // Whether to include results that are marked @inlinable or @usableFromInline.
+  /// Whether to include results that are marked @inlinable or @usableFromInline.
   IncludeUsableFromInline = 1 << 2,
+  /// Disable macro expansions.
+  DisableMacroExpansions = 1 << 3,
```

Comment:
These "disable macro expansions" flags end up in a lot of places. Are they all necessary, or can some be re-derived from ASTScope and location info?

Reply:
The need for `DisableMacroExpansions` is all related to default literal type lookup. Default literal type lookup tries to find a literal type name in the module scope, which, without the added flag, triggers top-level macro expansions. `DefaultTypeRequest` also doesn't have source location to check AST scope information, so we'd have to plumb through the locations from every call site, which I did give a try. That added location, however, caused default literal type lookup to look into type members, at which point it felt like opening a can of worms and it was no simpler than adding `DisableMacroExpansions`.